### PR TITLE
Check 'Docker' and 'Docker Compose' already installed

### PR DESCRIPTION
**Pull request by lucydodo**
This PR related to issue #68.

Check if 'Docker' and 'Docker Compose' are installed on user system,
and if so skip the component installation step and start the Redash installation process.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -16,8 +16,15 @@ if [ "0$ID" -ne 0 ]
...
 fi
 
-# Ensure the script is running on something it can work with
-if [ ! -f /etc/os-release ]; then
+# Ensure the 'docker' and 'docker-compose' commands are available
+# and if not, ensure the script can install them
+SKIP_DOCKER_INSTALL=no
+if [ -x "$(command -v docker)" ]; then
+  # The first condition is 'docker-compose (v1)' and the second is 'docker compose (v2)'.
+  if [ -x "$(command -v docker-compose)" ] || (docker compose 1> /dev/null 2>& 1 && [ $? -eq 0 ]); then
```
Comment (justinclift) [inline setup.sh:24]: You could probably improve this check a bit by combining the two checks into one.

Something like this:

```
if [ -x "..test1.." -a -x "..test2.." ]; then ...
```

The concept above is the `-a` (AND) can be use to join tests:  "if A ***AND*** B are both true".

There's also the `-o` (OR) operator, for testing if any of a series of tests is true, but that's probably not useful for this PR.

Comment (lucydodo) [reply setup.sh:24]: That was my first thought, but wouldn't conditional statements be too complicated?
And the second if statement is using an OR condition to check for both docker-compose v1 and v2.

Comment (justinclift) [reply setup.sh:24]: Good point.  Yeah, complication just leads to mistakes.  Let's go with your approach, as it should mean future people can maintain it pretty easily too. :smile:

Comment (lucydodo) [reply setup.sh:24]: That's great, ~~can we merge now? :)~~ Nah.

File: README.md [hunk 2]
```diff
@@ -33,6 +33,10 @@ This script should be run as the `root` user on a supported Linux system (as per
 When run, the script will install the needed packages (mostly Docker) then install Redash, ready for you to configure
 and begin using.
 
+> [!TIP]
+> If you are not on a supported Linux system, you can manually install the 'docker' and 'docker compose',
```
Comment (justinclift) [inline README.md]: Probably drop the `the`. :smile:

Comment (lucydodo) [reply README.md]: Good catch. Dropped.

File: setup.sh (hunk 1 above)
Comment (justinclift) [reply setup.sh:24]: Now we can merge. :smile:

**Review by justinclift — APPROVED** [review]
Looks good to me. :smile:

**Comment by lucydodo** [comment]
Thank you for reviewing this. :)

**Pull request merged**